tensorboard/plugins/profile/memory_viewer/mv_line_chart/mv-line-chart.ts

```diff
@@ -0,0 +1,259 @@
+      notify:true,
+    },
+    spanPlot: {
+      type:Object,
+      notify:true,
+    },
+    colorScale: {
+      type:Object,
+      notify:true,
+    },
+    _selectedEntityInSrcChart: {
+      type:Object,
+      notify:true,
+      observer: '_selectedEntityChanged',
+    },
+    _selectedEntityInDstChart: {
+      type:Object,
+      notify:true,
+      observer: '_selectedEntityChanged',
+    },
+  },
+  _makeChartDataset() {
+    if (!this.data) { return;}
+    this.bufferSizes = this.data.heapSizes.map((val, index) => [index, val]);
+    this.unpaddedBufferSizes = this.data.unpaddedHeapSizes.map((val, index) =>
+                                                               [index, val]);
+    let maxHeap = this.data.maxHeap;
+    this.data.maxHeap.reduce(function(sum, item, i) {
+        maxHeap[i]['offset'] = sum;
+        return sum + item.sizeMiB;
+      }, 0);
+    this.maxHeap = maxHeap;
+    let maxHeapBySize = this.data.maxHeapBySize;
+    this.data.maxHeapBySize.reduce(function(sum, item, i) {
+        maxHeapBySize[i]['offsetBySize'] = sum;
+        return sum + item.sizeMiB;
+      }, 0);
+    this.maxHeapBySize = maxHeapBySize;
+  },
+  /**
+   * Draw heap memory allocation line charts in program order. We also draw the
+   * breakdown of peak heap allocation in program order and by size. The
+   * position of the peak heap allocation is annotated on the graph. And an
+   * additional span is plotted to indicate the life range of the logical buffer
+   * allocation.
+   */
+  _drawProgramOrder() {
+    if (!this.data) { return;}
+    let xScale = new Plottable.Scales.Linear();
+    let yScale = new Plottable.Scales.Linear();
+    let xAxis = new Plottable.Axes.Numeric(xScale, "bottom");
+    let yAxis = new Plottable.Axes.Numeric(yScale, "left");
+
+    // Draw buffer sizes line.
+    let bufferSizesLine = new Plottable.Plots.Line();
+    bufferSizesLine.addDataset(new Plottable.Dataset(this.bufferSizes));
+    bufferSizesLine.x(function(d) {return d[0]; }, xScale)
+                   .y(function(d) {return d[1]; }, yScale)
+                   .attr("stroke", "#888888");
+
+    let unpaddedBufferSizesLine = new Plottable.Plots.Line();
+    unpaddedBufferSizesLine.addDataset(
+        new Plottable.Dataset(this.unpaddedBufferSizes));
+    unpaddedBufferSizesLine.x(function(d) {return d[0]; }, xScale)
+                           .y(function(d) {return d[1]; }, yScale)
+                           .attr("stroke", "#DE4406");
+
+    // Draw a band of width 10 to indicate the peak heap size location.
+    let bandPlot = new Plottable.Plots.Rectangle();
+    bandPlot.addDataset(
+        new Plottable.Dataset(
+          [this.bufferSizes[this.data.peakHeapSizePosition]]));
+    bandPlot.x(function(d) { return d[0]; }, xScale)
+            .y(function(d) { return 0; }, yScale)
+            .x2(function(d) { return d[0] + 10; })
+            .y2(function(d) { return d[1]; })
+            .attr("fill", "#DE4406")
```

Comment:
Might be nice to factor out some constants for `#DE4406` and `#888888` since they're referenced in a few places.

Reply:
Thanks Nick. I found that I can just use the built-in red/grey color. 
Sure, I will add a demo json in a separate PR.